subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan_aux.c

```diff
@@ -0,0 +1,345 @@
+	radio_gpio_pa_lna_enable(remainder_us +
+				 radio_rx_ready_delay_get(lll->phy, 1) -
+				 CONFIG_BT_CTLR_GPIO_LNA_OFFSET);
+#else /* !CONFIG_BT_CTLR_PHY */
+	radio_gpio_pa_lna_enable(remainder_us +
+				 radio_rx_ready_delay_get(0, 0) -
+				 CONFIG_BT_CTLR_GPIO_LNA_OFFSET);
+#endif /* !CONFIG_BT_CTLR_PHY */
+#endif /* CONFIG_BT_CTLR_GPIO_LNA_PIN */
+
+#if defined(CONFIG_BT_CTLR_PROFILE_ISR) || \
+	defined(CONFIG_BT_CTLR_GPIO_PA_PIN)
+	radio_tmr_end_capture();
+#endif /* CONFIG_BT_CTLR_PROFILE_ISR */
+
+	/* scanner always measures RSSI */
+	radio_rssi_measure();
+
+#if defined(CONFIG_BT_CTLR_XTAL_ADVANCED) && \
+	(EVENT_OVERHEAD_PREEMPT_US <= EVENT_OVERHEAD_PREEMPT_MIN_US)
+	/* check if preempt to start has changed */
+	if (lll_preempt_calc(evt, (TICKER_ID_SCAN_AUX_BASE +
+				   ull_scan_aux_lll_handle_get(lll)),
+			     ticks_at_event)) {
+		radio_isr_set(isr_done, lll);
+		radio_disable();
+	} else
+#endif /* CONFIG_BT_CTLR_XTAL_ADVANCED */
+	{
+		uint32_t ret;
+
+		ret = lll_prepare_done(lll);
+		LL_ASSERT(!ret);
+	}
+
+	DEBUG_RADIO_START_O(1);
+
+	return 0;
+}
+
+static void abort_cb(struct lll_prepare_param *prepare_param, void *param)
+{
+	int err;
+
+	/* NOTE: This is not a prepare being cancelled */
+	if (!prepare_param) {
+		/* Perform event abort here.
+		 * After event has been cleanly aborted, clean up resources
+		 * and dispatch event done.
+		 */
+		radio_isr_set(isr_done, param);
+		radio_disable();
+		return;
+	}
+
+	/* NOTE: Else clean the top half preparations of the aborted event
+	 * currently in preparation pipeline.
+	 */
+	err = lll_hfclock_off();
+	LL_ASSERT(!err || err == -EBUSY);
+
+	lll_done(param);
+}
+
+static void isr_rx(void *param)
+{
+	struct lll_scan_aux *lll = param;
+	uint8_t rssi_ready;
+	uint8_t trx_done;
+	uint8_t crc_ok;
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+		lll_prof_latency_capture();
+	}
+
+	/* Read radio status and events */
+	trx_done = radio_is_done();
+	if (trx_done) {
+		crc_ok = radio_crc_is_valid();
+		rssi_ready = radio_rssi_is_ready();
+	} else {
+		crc_ok = rssi_ready = 0U;
+	}
+
+	/* Clear radio rx status and events */
+	lll_isr_rx_status_reset();
+
+	/* No Rx */
+	if (!trx_done) {
+		/* TODO: Combine the early exit with above if-then-else block
+		 */
+		goto isr_rx_do_close;
+	}
+
+	if (crc_ok) {
+		int err;
+
+		err = isr_rx_pdu(lll, rssi_ready);
+		if (!err) {
+			if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+				lll_prof_send();
+			}
+
+			return;
+		}
+	}
+
+isr_rx_do_close:
+	radio_isr_set(isr_done, lll);
+	radio_disable();
+}
+
+static int isr_rx_pdu(struct lll_scan_aux *lll, uint8_t rssi_ready)
+{
+	struct node_rx_pdu *node_rx;
+	struct node_rx_ftr *ftr;
+	struct pdu_adv *pdu;
+
+	node_rx = ull_pdu_rx_alloc_peek(3);
+	if (!node_rx) {
+		return -ENOBUFS;
+	}
+
+	pdu = (void *)node_rx->pdu;
+	if ((pdu->type != PDU_ADV_TYPE_EXT_IND) || !pdu->len) {
+		return -EINVAL;
+	}
+
+	ull_pdu_rx_alloc();
+
+	trx_cnt++;
+
+	switch (lll->phy) {
+	case BIT(0):
```

Comment:
Ideally we would use the PHY_1M, PHY_2M and PHY_CODED definitions from pdu.h here